src/Ticket.php

```diff
@@ -1747,8 +1747,8 @@ public function prepareInputForAdd($input)
                 foreach ($items as $items_id) {
                     if ($item = getItemForItemtype($itemtype)) {
                         $item->getFromDB($items_id);
-                        $input['_states_id_of_item']    = $item->fields['states_id'];
-                        $input['_locations_id_of_item'] = $item->fields['locations_id'];
+                        $input['_states_id_of_item']    = $item->fields['states_id'] ?? 0;
+                        $input['_locations_id_of_item'] = $item->fields['locations_id'] ?? 0;
```

Comment:
@cedric-anne is that what you suggest?
```suggestion
                        $input['_states_id_of_item']    = $item->fields['states_id'] ?? null;
                        $input['_locations_id_of_item'] = $item->fields['locations_id'] ?? null;
```

Reply:
Yes, but as we do not have a test case for this, I am not sure it will work as expected.

If there is no such field for an item, but the rule says that the location of the item should be used, should we set the location to 0, or should be preserve the existing location value ?

Reply:
Ping @tsmr .